src/Sparrow/Json/JsonContextPoolBase.cs

```diff
@@ -348,12 +417,27 @@ public void Dispose()
             {
                 if (_disposed)
                     return;
+
+#if DEBUG
+                JsonContextPoolsHolder.AllContexts.TryRemove(this); 
+#endif
                 _cts.Cancel();
                 _disposed = true;
                 ThreadLocalCleanup.ReleaseThreadLocalState -= CleanThreadLocalState;
                 _nativeMemoryCleaner.Dispose();
+
+#if Debug
+                var z = new HashSet<ContextStack>();
```

Comment:
If we want to keep it then `z` is probably not the best name :)

Reply:
yes